From 3c180f44aab6739f025095fd450c80538bcb1256 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Wed, 27 Jul 2011 14:24:30 +0800
Subject: [PATCH 149/259] ENGR00153764 ipu lib: add dummy ipu lib for imx6q

add dummy ipu lib for imx6q to pass vpu unit test building.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 ipu/Makefile               |  6 +++-
 ipu/mxc_ipu_hl_lib_dummy.c | 72 ++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 77 insertions(+), 1 deletion(-)
 create mode 100644 ipu/mxc_ipu_hl_lib_dummy.c

diff --git a/ipu/Makefile b/ipu/Makefile
index cd842ab..2921216 100644
--- a/ipu/Makefile
+++ b/ipu/Makefile
@@ -1,10 +1,14 @@
 CC=$(CROSS_COMPILE)gcc
 AR=$(CROSS_COMPILE)ar
 
+ifeq ($(PLATFORM), IMX6Q)
+OBJS = mxc_ipu_hl_lib_dummy.o
+else
 OBJS = mxc_ipu_hl_lib.o mxc_ipu_lib.o
+endif
 
 # list of platforms which want this test case
-INCLUDE_LIST:= IMX37_3STACK IMX51 IMX53
+INCLUDE_LIST:= IMX37_3STACK IMX51 IMX53 IMX6Q
 
 LIBNAME=libipu
 
diff --git a/ipu/mxc_ipu_hl_lib_dummy.c b/ipu/mxc_ipu_hl_lib_dummy.c
new file mode 100644
index 0000000..85f7933
--- /dev/null
+++ b/ipu/mxc_ipu_hl_lib_dummy.c
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.
+ *
+ */
+
+/*
+ * The code contained herein is licensed under the GNU Lesser General
+ * Public License.  You may obtain a copy of the GNU Lesser General
+ * Public License Version 2.1 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/lgpl-license.html
+ * http://www.gnu.org/copyleft/lgpl.html
+ */
+
+/*!
+ * @file mxc_ipu_hl_lib.c
+ *
+ * @brief IPU high level library implementation
+ *
+ * @ingroup IPU
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdint.h>
+#include <malloc.h>
+#include <string.h>
+#include <signal.h>
+#include <pthread.h>
+#include <time.h>
+#include <semaphore.h>
+#include <sys/time.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <linux/videodev.h>
+#include "mxc_ipu_hl_lib.h"
+
+int mxc_ipu_lib_task_init(ipu_lib_input_param_t * input,
+		ipu_lib_overlay_param_t * overlay,
+		ipu_lib_output_param_t * output,
+		int mode, ipu_lib_handle_t * ipu_handle)
+{
+	return 0;
+}
+
+void mxc_ipu_lib_task_uninit(ipu_lib_handle_t * ipu_handle)
+{
+	return;
+}
+
+int mxc_ipu_lib_task_buf_update(ipu_lib_handle_t * ipu_handle,
+	dma_addr_t new_inbuf_paddr, dma_addr_t new_ovbuf_paddr,
+	dma_addr_t new_ovbuf_alpha_paddr, void (output_callback)(void *, int),
+	void * output_cb_arg)
+{
+	return 0;
+}
+
+int mxc_ipu_lib_task_control(int ctl_cmd, void * arg, ipu_lib_handle_t * ipu_handle)
+{
+	return 0;
+}
+
+int mxc_ipu_lib_ipc_init(void)
+{
+	return 0;
+}
-- 
1.8.0

